File: design/src/commonMain/kotlin/dev/helw/playground/sdui/design/core/SizeToken.kt
/**
 * Copyright (c) 2024 Ahmed El-Helw and Abdulahi Osoble
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package dev.helw.playground.sdui.design.core

import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

enum class SizeToken(val underlyingSize: Dp) {
    SMALL(24.dp),
    MEDIUM(32.dp),
    LARGE(48.dp)
}

File: catalog-ios/Design/Core/SizeToken.swift
//
//  SizeToken.swift
//  Catalog
//
//  Created by Ahmed El-Helw on 9/9/23.
//

import Foundation

enum SizeToken : String, Codable {
  case small
  case medium
  case large
}

extension SizeToken {
  var value: CGFloat {
    switch self {
      case .small:
        24
      case .medium:
        32
      case .large:
        48
    }
  }
}
